### PR TITLE
add skeleton flask and supplier class

### PR DESCRIPTION
**Pull request by HughHuang96824**
The flask file is the base for this project. Please check this skeleton version. Also leave your comment on the supplier model.

### CODE REVIEW

File: Application/app.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Create Flask application
+######################################################################
+app = Flask(__name__)
+app.logger.setLevel(logging.INFO)
+
+######################################################################
+# Application Routes
+######################################################################
+@app.route("/")
+def index():
```
Comment (zhengqixi) [inline Application/app.py:21]: Would it be difficult to add typing here as well? Looks like flasks just wants a tuple as a return type.

File: Objects/supplier.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import json
+from typing import Dict
+#from product.py import Product
+
+class Apple:
```
Comment (zhengqixi) [inline Objects/supplier.py:5]: Apple seems a bit specific. We should probably have a more generic base class name like `Product`

File: Objects/supplier.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    @address.setter
+    def address(self, address: str) -> str:
+        self._address = address
+
+    @products.setter
+    def products(self, products: Dict[str, Apple]) -> None:
+        self._products = products
+
+    def toJSON(self) -> str:
+        return json.dumps(self, default = lambda o: {k.lstrip('_'): v for k, v in vars(o).items()}, indent = 4)
```
Comment (zhengqixi) [inline Objects/supplier.py:70]: This line seems a bit long. Can you run this through an autoformatter?

File: Objects/supplier.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+if __name__ == "__main__":
```
Comment (zhengqixi) [inline Objects/supplier.py:72]: Will probably need to come back to this and update to use `nose` or something at some point.

File: Application/app.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+######################################################################
+DEBUG = os.getenv("DEBUG", "False") == "True"
+PORT = os.getenv("PORT", "5000")
+
+######################################################################
+# Create Flask application
+######################################################################
+app = Flask(__name__)
+app.logger.setLevel(logging.INFO)
+
```
Comment (zhengqixi) [inline Application/app.py:16]: As part of the AC, we wanted to have a top level storage object to store suppliers created from `Create` calls, and to list them out later. Can we get a simple dictionary or something here for now?

File: Objects/supplier.py (hunk 2 above)
Comment (HughHuang96824) [reply Objects/supplier.py:5]: This is just for testing. The product class is written by Ji

File: Application/app.py (hunk 5 above)
Comment (HughHuang96824) [reply Application/app.py:16]: Sure

File: Objects/supplier.py (hunk 2 above)
Comment (zhengqixi) [reply Objects/supplier.py:5]: Cool. Then @JiA1996 can you update this code as appropriate with your PR?

Comment (JiA1996) [reply Objects/supplier.py:5]: ok

File: Objects/supplier.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+class Apple:
+    def __init__(self, id) -> None:
+        self.name = "apple"
+        self.index = str(id)
+
+    def getId(self):
+        return self.index
+
+class Supplier:
+    def __init__(self, fname: str, lname: str, age: int = None, address: str = None) -> None:
```
Comment (zhengqixi) [inline Objects/supplier.py:14]: Just `name`

**Pull request merged**